corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -989,6 +989,14 @@ def get_deleted_case_ids_in_domain(domain):
     def get_case_ids_in_domain_by_owners(domain, owner_ids, closed=None):
         return CaseAccessorSQL._get_case_ids_in_domain(domain, owner_ids=owner_ids, is_closed=closed)
 
+    @staticmethod
+    def iter_case_ids_by_domain_and_type(domain, type_=None):
+        from corehq.sql_db.util import run_query_across_partitioned_databases
+        q_expr = Q(domain=domain) & Q(type=type_)
```

Comment:
if `type_ is None` then the filter should be excluded. Should include `deleted=False`